File: tests/test_random_agent.py
import gym
import time
import numpy as np
from src.agents import RandomAgent

env = gym.make('LunarLander-v2')
agent  = RandomAgent(env.action_space.n)

episodes = 10000
rewards = []

print('---')
print('Running {} episodes in Python...'.format(episodes))
start = time.time()

for _ in range(episodes):
    env.reset()
    total_reward = 0
    while True:
        action = agent.act()
        obs, reward, done, info = env.step(action)
        total_reward += reward
        if done:
            rewards.append(total_reward)
            break
env.close()

end = time.time()
print('It took {:.2f} seconds to finish.'.format(end - start))
print('Average total reward: {:.2f}'.format(np.mean(rewards)))
print('---')

File: tests/testRandomAgent.ts
import GymieClient from '../src/client/GymieClient'
import { Continuous, Discrete } from '../src/client/Env'
import { Dict } from '../src/client/utils'

const wsApi = 'http://0.0.0.0:5000/gym'
const envId = 'LunarLander-v2'

function randomInt(min: number, max: number): number {
  return Math.floor(Math.random()*(max - min) + min)
}

(async () => {
  const gymie = new GymieClient()
  await gymie.connect(wsApi)

  const env = await gymie.make<Continuous, Discrete>(envId)
  const space = await env.action_space()

  const nActions = space.n
  const episodes = 1000

  console.log('---')
  console.log(`Running ${episodes} episodes in Javascript...`)

  const start = new Date().getTime()

  let i = episodes

  await (async function forLoop() {

    await env.reset()
    let totalReward = 0

    await (async function whileLoop() {
      const action = randomInt(0, nActions)
      const [obs, reward, done, info] = await env.step(action)
      totalReward += reward

      if (done) return
      await whileLoop()
    })()


    if (--i === 0) return
    await forLoop()
  })()

  await env.close()

  const end = new Date().getTime()
  console.log(`It took ${(end - start)/1000} seconds to finish.`)
  console.log('---')

  gymie.close()

})()
